sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/ScalaReflection.scala

```diff
@@ -329,35 +329,19 @@ object ScalaReflection extends ScalaReflection {
         }
         UnresolvedMapObjects(mapFunction, getPath, Some(cls))
 
-      case t if t <:< localTypeOf[Map[_, _]] =>
+      case t if t <:< localTypeOf[Map[_, _]] || t <:< localTypeOf[java.util.Map[_, _]] =>
```

Comment:
we should handle java map in `JavaTypeInference`, but I think it's better to do it in another PR and focus on scala map in this PR.